addons/sale_stock/sale_stock.py

```diff
@@ -389,7 +389,7 @@ def _get_master_data(self, cr, uid, move, company, context=None):
 
     def _get_invoice_line_vals(self, cr, uid, move, partner, inv_type, context=None):
         res = super(stock_move, self)._get_invoice_line_vals(cr, uid, move, partner, inv_type, context=context)
-        if move.procurement_id and move.procurement_id.sale_line_id:
+        if move.procurement_id and move.procurement_id.sale_line_id and inv_type in ('out_invoice', 'out_refund'):
```

Comment:
Maybe even faster if you put the check for inv_type first. 
out_refund should be there in _get_master_data then also?